diff --git a/node_modules/@types/promise/setimmediate/rejection-tracking.d.ts b/node_modules/@types/promise/setimmediate/rejection-tracking.d.ts
new file mode 100644
index 0000000..e436fa1
--- /dev/null
+++ b/node_modules/@types/promise/setimmediate/rejection-tracking.d.ts
@@ -0,0 +1,7 @@
+export function disable(): void
+export function enable(options: {
+    allRejections?: boolean,
+    whitelist?: Error[],
+    onUnhandled?: (displayId: number, error: Error) => void
+    onHandled?: (displayId: number, error: Error) => void
+}): void
diff --git a/node_modules/@types/promise/types-metadata.json b/node_modules/@types/promise/types-metadata.json
index 636265a..9ab13ba 100644
--- a/node_modules/@types/promise/types-metadata.json
+++ b/node_modules/@types/promise/types-metadata.json
@@ -16,7 +16,8 @@
     ],
     "declaredModules": [],
     "files": [
-        "index.d.ts"
+        "index.d.ts",
+        "setimmediate/rejection-tracking.d.ts"
     ],
     "hasPackageJson": false,
     "contentHash": "068b30ccfabfc941acd1c7d983171868fc516ce405e22f15c79081c4f324d830"
